### ISSUE TITLE
Need Fluent-Bit to support reading cert from Windows Certstore

### ISSUE THREAD

**Issue by duj4** [fluent/fluent-bit#9215]
**Is your feature request related to a problem? Please describe.**
This is not related to a problem but just a feature request.

**Describe the solution you'd like**
In our windows server, the cert is auto-enrolled when the server was provisioned and the private key export is disabled as well for the sake of security. In such case, the only way to refer to the cert is from Windows cert store, but it seems Fluent-Bit does not support this (Fluentd does, https://docs.fluentd.org/output/forward#how-to-connect-to-a-tls-ssl-enabled-server-with-windows-certstore-certificate). Therefore, could it be possible to add this to the existing [OUTPUT] plugins that support TLS in Windows version?

**Describe alternatives you've considered**
We have to generated cert and key separately manually, but there are hundreds of servers.

**Additional context**
There are hundreds of Windows servers in our production environment and we have to turn on TLS while sending the logs to Loki. If it does support reading cert from Windows cert store, we have to generate a separate cert/key pair manually (though we are trying to work out a programmatic way for this).

**Comment by cosmo0920** [comment]
We already read certificates from Root certstore on Windows. You mean that it should be configurable?
I wrote a mechanism from reading certstore on Fluentd side.

**Comment by cosmo0920** [comment]
> Windows FluentBit cannot refer to the cert from certstore but only read from the file.

For sure, it's not partially correct. This is because we already read Root certstore on Windows here: https://github.com/fluent/fluent-bit/blob/master/src/tls/openssl.c#L249
This could be used for reading system certificates.
To be configurable, we need to provide an interface to specify Windows specific parameters.

**Comment by duj4** [comment]
> > Windows FluentBit cannot refer to the cert from certstore but only read from the file.
> 
> For sure, it's not partially correct. This is because we already read Root certstore on Windows here: https://github.com/fluent/fluent-bit/blob/master/src/tls/openssl.c#L249 This could be used for reading system certificates. To be configurable, we need to provide an interface to specify Windows specific parameters.

Yes, that'd be better if there is any interface for the configuration, thanks @cosmo0920

**Comment by edsiper** [comment]
so it seems we are missing the config aspect

note: moving to another milestone

**Comment by cosmo0920** [comment]
I sent a [PR](https://github.com/fluent/fluent-bit/pull/10590) for implementing this.

This will be released as 4.1 under the current plan of development cycle.

**Comment by axlexn** [comment]
@cosmo0920 Thank you for implementing this! Do you think we also need `tls_cert_thumbprint`, as with Fluentd? I am not sure the certificate actually gets loaded with its corresponding key by providing a certstore name in `tls.windows.certstore_name` in 4.1. Adding a thumbprint parameter would also allow for selecting which certificate to use if there are multiple available.
